api/client/fip.go

```diff
@@ -104,11 +104,30 @@ func (cli *DockerCli) CmdFipRelease(args ...string) error {
 // Usage: docker fip attach [OPTIONS] <FIP> <CONTAINER>
 func (cli *DockerCli) CmdFipAttach(args ...string) error {
 	cmd := Cli.Subcmd("fip attach", []string{"FIP CONTAINER"}, "Connects a container to a floating IP", false)
+	flForce := cmd.Bool([]string{"f", "-force"}, false, "Deattach that FIP and attach it to this container")
 	cmd.Require(flag.Min, 2)
 	if err := cmd.ParseFlags(args, true); err != nil {
 		return err
 	}
-	return cli.client.FipAttach(context.Background(), cmd.Arg(0), cmd.Arg(1))
+	ctx := context.Background()
+	if *flForce {
+		filter, _ := filters.FromParam("dangling=false")
+		options := types.NetworkListOptions{
+			Filters: filter,
+		}
+
+		fips, err := cli.client.FipList(ctx, options)
+		if err != nil {
+			return err
+		}
+		for _, fip := range fips {
+			if fip["fip"] == cmd.Arg(0) {
+				cli.client.FipDetach(ctx, fip["container"])
+				break
+			}
+		}
+	}
+	return cli.client.FipAttach(ctx, cmd.Arg(0), cmd.Arg(1))
```

Comment:
How about do like this:

- try attach, return if ok
- if error is occupied, detach it, fail if err
- attach again

Three API invocation if occupied, same as proposed method. But if it is idle, only one API call is made.